sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { generate_uuid } from "rhea-promise";
+import { isTokenCredential, TokenCredential } from "@azure/core-auth";
+import {
+  ServiceBusClientOptions,
+  createConnectionContextForTokenCredential,
+  createConnectionContextForConnectionString
+} from "./constructorHelpers";
+import { ConnectionContext } from "./connectionContext";
+import { ClientEntityContext } from "./clientEntityContext";
+import { ClientType } from "./client";
+import { Sender } from "./sender";
+import { GetSessionReceiverOptions, ContextWithSettlement } from "./models";
+import { Receiver, ReceiverImpl, SubscriptionRuleManagement } from "./receivers/receiver";
+import { SessionReceiver, SessionReceiverImpl } from "./receivers/sessionReceiver";
+
+export class ServiceBusClient {
+  private _connectionContext: ConnectionContext;
+
+  /**
+   *
+   * @param connectionString A connection string for Azure Service Bus.
+   * NOTE: this connection string can contain an EntityPath, which is ignored.
+   * @param options Options for the service bus client.
+   */
+  constructor(connectionString: string, options?: ServiceBusClientOptions);
+  /**
+   *
+   * @param host The hostname of your Azure Service Bus.
+   * @param tokenCredential A valid TokenCredential for Service Bus or a
+   * Service Bus entity.
+   * @param options Options for the service bus client.
+   */
+  constructor(
+    hostName: string,
+    tokenCredential: TokenCredential,
+    options?: ServiceBusClientOptions
+  );
+  constructor(
+    connectionStringOrHostName1: string,
+    tokenCredentialOrServiceBusOptions2?: TokenCredential | ServiceBusClientOptions,
+    options3?: ServiceBusClientOptions
+  ) {
+    if (isTokenCredential(tokenCredentialOrServiceBusOptions2)) {
+      const hostName: string = connectionStringOrHostName1;
+      const tokenCredential: TokenCredential = tokenCredentialOrServiceBusOptions2;
+      const options: ServiceBusClientOptions | undefined = options3;
+
+      this._connectionContext = createConnectionContextForTokenCredential(
+        tokenCredential,
+        hostName,
+        options
+      );
+    } else {
+      const connectionString: string = connectionStringOrHostName1;
+      const options: ServiceBusClientOptions | undefined = tokenCredentialOrServiceBusOptions2;
+
+      this._connectionContext = createConnectionContextForConnectionString(
+        connectionString,
+        options
+      );
+    }
+  }
+
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param queueName The name of the queue to receive from.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  getReceiver(queueName: string, receiveMode: "peekLock"): Receiver<ContextWithSettlement>;
```

Comment:
No options?

Reply:
There aren't any at the moment - we can add if we expose something interesting there.